test/test_runs.py

```diff
@@ -96,117 +126,173 @@ def test_run_00_cli(capsys: Any):
     assert exc.type == SystemExit
     assert exc.value.code != 0
 
-    sys.argv = sargs
 
+@insulate_globals
+@pytest.mark.parametrize(
+    'name,gport',
+    [
+        (NAME, GPORT),
+        (NAME2, GPORT2),
+    ],
+)
+def test_run_02_delete(capsys: Any, name: str, gport: int):
+    '''Delete a project.
+
+    In this case the project should not already exist.
 
-def test_run_01_delete(capsys: Any):
-    'start delete'
-    sargs = sys.argv
+    This cleans up everything for the subsequent tests.
+
+    Args:
+        capsys: Pytest fixture for capturing stdout/stderr.
+        name: The grape project name for this test.
+        gport: The grafana port for this test.
+    '''
+    namegr, namepg, namezp = make_names(name)
     fct = inspect.stack()[0].function
 
     # Delete the primary set of resources.
-    sys.argv = [fct, '-v', '-n', NAME, '-g', str(GPORT)]
-    delete.main()
-    out, err = capsys.readouterr()
-    print(f'out=<<<{out}>>>')
-    print(f'err=<<<{err}>>>')
-    assert not os.path.exists(NAMEPG)
-    assert NAMEGR in err
-    assert NAMEPG in err
-    if os.path.exists(NAMEZP):
-        os.unlink(NAMEZP)
-    assert not os.path.exists(NAMEZP)
-
-    # Now delete the second set of recources.
-    sys.argv = [fct, '-v', '-n', NAME2, '-g', str(GPORT2)]
+    sys.argv = [fct, '-v', '-n', name, '-g', str(gport)]
     delete.main()
     out, err = capsys.readouterr()
     print(f'out=<<<{out}>>>')
     print(f'err=<<<{err}>>>')
-    assert not os.path.exists(NAMEPG2)
-    assert NAMEGR2 in err
-    assert NAMEPG2 in err
-    if os.path.exists(NAMEZP2):
-        os.unlink(NAMEZP2)
-    assert not os.path.exists(NAMEZP2)
-    sys.argv = sargs
-
-
-def test_run_02_create(capsys: Any):
-    'create'
-    sargs = sys.argv
+    assert not os.path.exists(namepg)
+    assert namegr in err
+    assert namepg in err
+    if os.path.exists(namezp):
+        os.unlink(namezp)
+    assert not os.path.exists(namezp)
+
+
+@insulate_globals
+@pytest.mark.parametrize(
+    'name,gport',
+    [
+        (NAME, GPORT),
+    ],
+)
+def test_run_03_create(capsys: Any, name: str, gport: int):
```

Comment:
These tests are quick, it might be worth considering making this into a fixture so that tests are decoupled and can be run in any order or individually.

Alternately, if that doesn't strike your fancy, may I suggest [pytest-dependency](https://github.com/RKrahl/pytest-dependency)?

Reply:
Those are both great suggestions. I am surprised that the tests are so quick now.

My original motivation for cascading them was that they were slow. However, i have been unable to duplicate the slowness so creating a fixture makes sense. If/when they slow down, we can always go back.

Reply:
Hmmm, i wonder if my little decorator is causing execution problems... Need to check that....

Reply:
The decorator was causing problems. The tests were not actually running or, apparently, linting. I removed the decorator. Everything is copacetic now but tests much longer to run.

I added comments to describe why it is organized the way it it.

Reply:
I am convinced by your arguments and have added pytest-depend dependencies. Thank you!